### PR TITLE
Add new dev cmds to Software CLI Reference guide

### PR DESCRIPTION
**Pull request by sunkickr**
This PR adds documentation for astro dev parse, pytest, and restart to the software CLI reference guide

### CODE REVIEW

File: software/cli-reference.md [hunk 1]
```diff
@@ -575,6 +577,26 @@ Run `astro dev logs [flags]` to start tracking logs for your Scheduler, Webserve
 | `--webserver` | None       | Outputs only Webserver logs.                       |
 | `--triggerer` | None       | Outputs only Triggerer logs.                       |
 
+## astro dev parse
+
+Parse the DAGs in a locally hosted Astro project to quickly check them for errors. For more information about testing DAGs locally, read [Run Tests with the Astro CLI](test-and-troubleshoot-locally.md#run-tests-with-the-astro-cli).
+
+:::info
+
+This command requires Astro Runtime version `4.1.0`+ or for `pytest` to be present in your `requirments.txt` file. For more information, see [Astro Runtime Release Notes](https://docs.astronomer.io/astro/runtime-release-notes#astro-runtime-410).
```
Comment (paolaperaza) [inline software/cli-reference.md:586]:
```suggestion
This command requires the [pytest](https://docs.pytest.org/en/7.1.x/contents.html) package. To install it to your Astro project, add `pytest` to your `requirements.txt` file.
```

@sunkickr Do we actually need `pytest` for this parse command? If so, I assume we want to scrap this Runtime note and just tell people to add `pytest`. And no expected issues on that with AC?

Comment (paolaperaza) [reply software/cli-reference.md:586]: Looks like this requirement isn't listed for Astro: https://docs.astronomer.io/astro/cli-reference/astrocloud-dev-parse

File: software/cli-reference.md [hunk 2]
```diff
@@ -583,6 +605,45 @@ Lists all running Docker containers for your local Airflow environment. This com
...
+
+Run unit tests for your data pipelines on Astro with `pytest`, a testing framework for Python. When you run this command, the Astro CLI creates a local Python environment that includes your DAG code, dependencies, and Astro Runtime Docker image. The CLI then runs any pytests in the `tests` directory of your Astro project and shows you the results of those tests in your terminal.
+
+For more information on this functionality, see [Test and Troubleshoot Locally](test-and-troubleshoot-locally.md).
+
+:::info
+
+This command requires Astro Runtime version `4.1.0`+ or for `pytest` to be present in your `requirments.txt` file. For more information, see [Astro Runtime Release Notes](https://docs.astronomer.io/astro/runtime-release-notes#astro-runtime-410).
+
+:::
```
Comment (paolaperaza) [inline software/cli-reference.md:618]: Same thing here

File: software/cli-reference.md [hunk 3]
```diff
@@ -583,6 +605,45 @@ Lists all running Docker containers for your local Airflow environment. This com
 
 `astro dev ps`
 
+## astro dev pytest
+
+Run unit tests for your data pipelines on Astro with `pytest`, a testing framework for Python. When you run this command, the Astro CLI creates a local Python environment that includes your DAG code, dependencies, and Astro Runtime Docker image. The CLI then runs any pytests in the `tests` directory of your Astro project and shows you the results of those tests in your terminal.
```
Comment (paolaperaza) [inline software/cli-reference.md:610]:
```suggestion
Run unit tests for your data pipelines with `pytest`, a testing framework for Python. When you run this command, the Astro CLI creates a local Python environment that includes your DAG code, dependencies, and Astronomer Certified Docker image. The CLI then runs any pytests in the `tests` directory of your Astro project and shows you the results of those tests in your terminal.
```

File: software/cli-reference.md [hunk 4]
```diff
@@ -583,6 +605,45 @@ Lists all running Docker containers for your local Airflow environment. This com
...
+
+### Flags
+
+| Flag          | Value Type | Usage                                              |
+| ------------- | ---------- | -------------------------------------------------- |
+| `--env`       | string     | The filepath to your environment variables. (The default is `.env`)  |
+
+## astro dev restart
+
+Stop your Airflow environment, rebuild your Astronomer project into a Docker image, and restart your Airflow environment with the new Docker image.
```
Comment (paolaperaza) [inline software/cli-reference.md:632]:
```suggestion
Stop your Airflow environment, rebuild your Astro project into a Docker image, and restart your Airflow environment with the new Docker image.
```

File: software/cli-reference.md [hunk 5]
```diff
@@ -583,6 +605,45 @@ Lists all running Docker containers for your local Airflow environment. This com
...
+
+This command can be used to rebuild an Astronomer project and run it locally. For more information, read [Develop and Run a Project Locally](develop-project.md#build-and-run-a-project-locally).
```
Comment (paolaperaza) [inline software/cli-reference.md:634]:
```suggestion
This command can be used to rebuild an Astro project and run it locally. For more information, read [Develop and Run a Project Locally](develop-project.md#build-and-run-a-project-locally).
```

File: software/cli-reference.md [hunk 6]
```diff
@@ -583,6 +605,45 @@ Lists all running Docker containers for your local Airflow environment. This com
...
+
+:::info
+
+This command requires Astro Runtime version `4.1.0`+ or for `pytest` to be present in your `requirments.txt` file. For more information, see [Astro Runtime Release Notes](https://docs.astronomer.io/astro/runtime-release-notes#astro-runtime-410).
+
+:::
+
+### Usage
+
+`astro dev pytest <pytest-filepath>` to run specfic Pytests on your DAGs. use `<pytest-filepath>` to specify a specific test.
```
Comment (paolaperaza) [inline software/cli-reference.md:622]:
```suggestion
`astro dev pytest <pytest-filepath>` to run specfic Pytests on your DAGs. Use `<pytest-filepath>` to specify a specific test. 
```

File: software/cli-reference.md [hunk 7]
```diff
@@ -583,6 +605,45 @@ Lists all running Docker containers for your local Airflow environment. This com
...
+### Usage
+
+`astro dev pytest <pytest-filepath>` to run specfic Pytests on your DAGs. use `<pytest-filepath>` to specify a specific test. 
+
+### Flags
+
+| Flag          | Value Type | Usage                                              |
+| ------------- | ---------- | -------------------------------------------------- |
+| `--env`       | string     | The filepath to your environment variables. (The default is `.env`)  |
+
```
Comment (paolaperaza) [inline software/cli-reference.md:629]: Any reason these flags are different than in Astro CLI docs? I know it's mentioned up top, but do we not add `<pytest-filepath>` too?

image

**Review by paolaperaza — CHANGES_REQUESTED** [review]
@sunkickr A few questions back to you actually on functionality/differences between Astro and Software on these - lemme know what you think and I'll take another pass

File: software/cli-reference.md (hunk 1 above)
Comment (sunkickr) [reply software/cli-reference.md:586]: if you have the right runtime image you don't need pytest. if you don't have the right runtime image you do need pytest

Comment (sunkickr) [reply software/cli-reference.md:586]: we added pytest to runtime so users would not need to add pytest to requirements

Comment (jwitz) [reply software/cli-reference.md:586]: Does the CLI have error handling for this? Ie if someone runs the command before Runtime 4.1.0, the CLI tells them what went wrong and how to get `pytest`? If so, we should remove this note and always assume that a user is running latest.

**Comment by jwitz** [comment]
@sunkickr I moved this PR to https://github.com/astronomer/docs/pull/759 because I can easily merge it into v0.28 versioned docs from there. Also made some quick changes here so that it's hopefully ready to merge

**Pull request closed without merging**